src/universal/Add Gamesync ID.c

```diff
@@ -0,0 +1,188 @@
+#include <pksm.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+/** checkOnlineID(version, saveData)
+ * Online IDs
+ *
+ * XY/ORAS
+ *  GameSync
+ *      offset: 0x14008
+ *      bytes: 8
+ *
+ * SM
+ *  GameSync
+ *      offset: 0x1210
+ *      bytes: 8
+ *  NexUniqueID
+ *      offset: 0x1218
+ *      bytes: 16
+ *  FestaID
+ *      offset: 0x1228
+ *      bytes: 4
+ *
+ * USUM
+ *  GameSync
+ *      offset: 0x1410
+ *      bytes: 8
+ *  NexUniqueID
+ *      offset: 0x1418
+ *      bytes: 16
+ *  FestaID
+ *      id[0:4]
+ *          offset: 0x1428
+ *          bytes: 4
+ *      id[5:11]
+ *          offset: 0x1418
+ *          bytes: 8
+ */
+int checkOnlineID(int version)
+{
+    int ofsSyncID = 0, sizeSyncID = 8, i;
+    switch (version)
```

Comment:
You can move this switch in its own function instead of duplicating this code where it is needed